cylc/flow/config_schema.py

```diff
@@ -0,0 +1,609 @@
+                'batch submit command template': [VDR.V_STRING]
+            },
+            'default directives': {
+                '__MANY__': [VDR.V_STRING],
+            },
+        },
+        '__MANY__': {
+            'run directory': [VDR.V_STRING, ''],
+            'work directory': [VDR.V_STRING, ''],
+            'suite definition directory': [VDR.V_STRING],
+            'task communication method': [VDR.V_STRING, ''],
+            'submission polling intervals': [VDR.V_STRING, ''],
+            'submission retry delays': [VDR.V_INTERVAL_LIST, None],
+            'execution polling intervals': [VDR.V_STRING, ''],
+            'scp command': [VDR.V_STRING, ''],
+            'ssh command': [VDR.V_STRING, ''],
+            'use login shell': [VDR.V_STRING, ''],
+            'login hosts': [VDR.V_STRING, ''],
+            'batch system': [VDR.V_STRING, ''],
+            'cylc executable': [VDR.V_STRING, ''],
+            'global init-script': [VDR.V_STRING, ''],
+            'copyable environment variables': [VDR.V_STRING, ''],
+            'retrieve job logs': [VDR.V_STRING, ''],
+            'retrieve job logs command': [VDR.V_STRING, ''],
+            'retrieve job logs max size': [VDR.V_STRING, ''],
+            'retrieve job logs retry delays': [VDR.V_STRING, ''],
+            'task event handler retry delays': [VDR.V_STRING, ''],
+            'tail command template': [VDR.V_STRING, ''],
+            'owner': [VDR.V_STRING_LIST],
+            'batch systems': {
+                'err tailer': [VDR.V_STRING, ''],
+                'out tailer': [VDR.V_STRING, ''],
+                'err viewer': [VDR.V_STRING, ''],
+                'out viewer': [VDR.V_STRING, ''],
+                'job name length maximum': [VDR.V_STRING, ''],
+                'execution time limit': [VDR.V_INTERVAL_LIST],
+                'execution polling intervals': [VDR.V_INTERVAL_LIST],
+                'execution retry delays': [VDR.V_INTERVAL_LIST],
+                'batch submit command template': [VDR.V_STRING]
+            },
+            'default directives': {
+                '__MANY__': [VDR.V_STRING],
+            },
+        },
+    },
+    'email': {
+        'mail events': [VDR.V_STRING_LIST, None],
+        'mail from': [VDR.V_STRING],
+        'mail smtp': [VDR.V_STRING],
+        'mail to': [VDR.V_STRING],
+        'mail footer': [VDR.V_STRING],
+        'monitor': {
+            'sort order': [VDR.V_STRING, 'definition', 'alphanumeric'],
+        },
+    },
+    'scheduling': {
+        'initial cycle point': [VDR.V_CYCLE_POINT],
+        'final cycle point': [VDR.V_STRING],
+        'initial cycle point constraints': [VDR.V_STRING_LIST],
+        'final cycle point constraints': [VDR.V_STRING_LIST],
+        'hold after point': [VDR.V_CYCLE_POINT],
+        'cycling mode': (
+            [VDR.V_STRING, Calendar.MODE_GREGORIAN] +
+            list(Calendar.MODES) + ["integer"]),
+        'runahead limit': [VDR.V_STRING],
```

Comment:
I've done it, but I think you underestimate the conservatism of our local user base.

Reply:
> I think we can finally drop the ancient runahead limit item

This might be a bit controversial, we haven't even deprecated `runahead limit` have we?

`max active cycle points` is not a direct replacement for `runahead limit`, there are reasons people might prefer this approach. I've done a quick scan through our currently running suites, 10% are using `runahead limit`.

Reply:
More context picking the first two suites off the list of `runahead limit` users:

1. Not so compelling but from a workflow standpoint P1D makes sense.

```ini
[scheduling]
    runahead limit = P3D
    [[dependencies]]
        [[[P1D]]]
        [[[T00]]]
        [[[T05]]]
```

2. Fairly compelling, it would require Jinja2 to calculate `max active cycles` in this setup:

```ini
[scheduling]
    runahead limit = P5D

[[dependencies]]
    [[[T00, T12]]]  # base workflow
{% if FOO %}
    [[[T06, T09, T18, T21]]]
{% endif %}
{% if BAR %}
    [[[T03, T09, T15, T21]]]
{% endif %}
```

Reply:
By drop I meant deprecate (and drop from the spec) as explained below.

But OK fair enough, sorry @wxtim.  Can you resolve this conversation after restoring `runahead limit`.

Reply:
This one might be worth discussing in more detail - bumped to #3405